website/www/site/content/en/documentation/runtime/environments.md

```diff
@@ -15,56 +15,168 @@ See the License for the specific language governing permissions and
 limitations under the License.
 -->
 
-# Container environments
+# Container Environments
 
-The Beam SDK runtime environment is isolated from other runtime systems because the SDK runtime environment is [containerized](https://s.apache.org/beam-fn-api-container-contract) with [Docker](https://www.docker.com/). This means that any execution engine can run the Beam SDK.
+The Beam SDK runtime environment is [containerized](https://www.docker.com/resources/what-container) with [Docker](https://www.docker.com/) to isolate it from other runtime systems. This means any execution engine can run the Beam SDK. To learn more about the container environment, read the Beam [SDK Harness container contract](https://s.apache.org/beam-fn-api-container-contract).
 
-This page describes how to customize, build, and push Beam SDK container images.
+Prebuilt SDK container images are released per supported language version during Beam releases and and pushed to [Docker Hub](https://hub.docker.com/search?q=apache%2Fbeam&type=image)
 
-Before you begin, install [Docker](https://www.docker.com/) on your workstation.
+## Custom Containers
 
-## Customizing container images
+Users may want to customize container images for many reasons, including:
 
-You can add extra dependencies to container images so that you don't have to supply the dependencies to execution engines.
+* pre-installing additional dependencies,
+* launching third-party software
+* further customizing the execution environment
 
-To customize a container image, either:
-* [Write a new](#writing-new-dockerfiles) [Dockerfile](https://docs.docker.com/engine/reference/builder/) on top of the original.
-* [Modify](#modifying-dockerfiles) the [original Dockerfile](https://github.com/apache/beam/blob/master/sdks/python/container/Dockerfile) and reimage the container.
+ This guide describes how to create and use customized containers for the Beam SDK.
 
-It's often easier to write a new Dockerfile. However, by modifying the original Dockerfile, you can customize anything (including the base OS).
+### Prerequisites
 
-### Writing new Dockerfiles on top of the original {#writing-new-dockerfiles}
+* You will need to have a version of the Beam SDK >= 2.21.0.
+* You will need to have [Docker installed](https://docs.docker.com/get-docker/).
```

Comment:
This is also a soft requirement - users could use CloudBuild or other equivalents that can build a container on demand from a given Dockerfile.